nemo/collections/tts/modules/transformer_dec24.py

```diff
@@ -0,0 +1,691 @@
+#############################################################################
+# Copyright (c) 2023 NVIDIA CORPORATION. All rights reserved.
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#############################################################################
+import math
+
+import torch
+import torch.nn.functional as F
+from flash_attn import flash_attn_varlen_kvpacked_func, flash_attn_varlen_qkvpacked_func
+from flash_attn.layers.rotary import RotaryEmbedding
+from torch import nn
+from torch.cuda import amp
+from torch.nn.utils.rnn import pad_sequence
+
+
+# alibi helper functions (start)
+def get_relative_positions(seq_len: int, device, symmetric: bool = True) -> torch.tensor:
+    x = torch.arange(seq_len)[None, :]
+    y = torch.arange(seq_len)[:, None]
+    rel_pos = (x - y).to(device)
+    if symmetric:
+        rel_pos = -rel_pos.abs()
+    return rel_pos
+
+
+def get_alibi_slope(num_heads):
+    x = (2**8) ** (1 / num_heads)
+    return torch.tensor([1 / x ** (i + 1) for i in range(num_heads)]).unsqueeze(-1).unsqueeze(-1)
+
+
+# alibi helper functions (end)
+
+
+class ConvNorm(torch.nn.Module):
```

Comment:
Should we remove this and use https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/modules/submodules.py#L140 instead?

Reply:
I renamed this class